### DISCUSSION TITLE
useFetch (or lazy variant) is firing only once

### DISCUSSION THREAD

**Discussion by arun-mani-j** [nuxt/nuxt#16936] [Questions] [answered]
Hii.
Sorry if it is a dupe.
To produce a minimal working example, install Nuxt 3 and change `app.vue` to:

```vue
<template>
  <div>
    Type something
    <input @change="fetchData" v-model="text" />
    {{ dataFetched }}
    <button @click="fetchData">Refresh</button>
  </div>
</template>

<script setup>
const text = ref(null);
const dataFetched = ref(null);

const fetchData = (e) => {
  console.log("Input changed to", text.value);
  const { data, error } = useLazyFetch(
    "https://random-data-api.com/api/name/random_name"
  );

  watch(error, (e) => {
    console.log(error);
  });
  watch(data, (newData) => {
    dataFetched.value = data;
  });
};
</script>
```

What I wish to do is to fetch data from an API whenever the input changes. If I change the input, for the first time it works. After that, even though the `dataFetched` function is triggered (as evident from the Console logs), the data is not refreshed. Even if I press the button it is not changing.

I'm pretty new to web development, so sorry if it is very newbish :sweat_smile: (As a side note, am I using the `error` or `watch` properly)

Thanks in advance!

**Comment by mihail727** [comment, marked as answer]
try it 
`await useFetch(url, {
   initialCache: false,
})`

**Comment by mihail727** [comment]
or u can set `key` field in `useFetch` like this:
```ts
await useFetch(url, {
    key: Date.now().toString()
})
```

**Reply by arun-mani-j** [reply]
Thanks for the help :)

I actually used `key: new Date().toISOString()` in my application and faced the issue. But when uploading the code here, I uploaded the wrong one (the example without use of key). 

However, a day later, after a full restart, the app started working fine. But why the code refused to work that day even when I rebuild multiple times remains a mystery ...

Can you also tell me if the way I'm checking for error and success is the right way?

Thanks again for your help ^^

**Reply by mihail727** [reply]
I haven't worked with useLazyFetch yet...
But everything seems to be ok...
You can also use the normal Promise methods like:
`useLazyFetch(...).catch((e) => e)` or `useLazyFetch(...).then((...) => ...)`
